logger.go

```diff
@@ -0,0 +1,21 @@
+package producer
+
+import (
+	"fmt"
+)
+
+// Logger represents a simple interface used by kinesis-producer to handle logging
+type Logger interface {
+	Info(msg string, values ...Value)
+	Error(msg string, err error, values ...Value)
```

Comment:
```suggestion
         Info(args ...interface{})
	Error(args ...interface{})
```
Why do you say about changing the interface signature to something like this? this will save the `Value` entry and the logrus/zap implementations.
You can even make it a field-logger by treat the `args` param as a list of key-value pairs.

Reply:
I personally don't like it as almost all loggers at least expect a string so taking interface{} where you know you need a string doesn't sound too good to me. Also, I think having a struct to define a key value pair is better than an even number of args and then grouping them inside the function. I think it's cleaner for both the API and the internal implementation.  I don't see much value in doing it but you are maintaining the code so I can do what makes your workflow better.